osu.Game/Users/UserPanel.cs

```diff
@@ -50,164 +34,64 @@ public class UserPanel : OsuClickableContainer, IHasContextMenu
 
         protected Action ViewProfile;
 
-        public UserPanel(User user)
+        protected DelayedLoadUnloadWrapper Background;
+
+        private SpriteIcon statusIcon;
+        private OsuSpriteText statusMessage;
+        private TextFlowContainer lastVisitMessage;
+
+        protected UserPanel(User user)
         {
             if (user == null)
                 throw new ArgumentNullException(nameof(user));
 
             User = user;
-
-            Height = height - status_height;
         }
 
-        [BackgroundDependencyLoader(permitNulls: true)]
-        private void load(UserProfileOverlay profile)
-        {
-            if (colours == null)
-                throw new InvalidOperationException($"{nameof(colours)} not initialized!");
+        [Resolved(canBeNull: true)]
+        private UserProfileOverlay profileOverlay { get; set; }
 
-            FillFlowContainer infoContainer;
+        [Resolved(canBeNull: true)]
+        private OverlayColourProvider colourProvider { get; set; }
+
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Masking = true;
+            BorderColour = colourProvider?.Light1 ?? colours.GreyVioletLighter;
```

Comment:
`UserListPanel` has a weird border rendering issue:

![image](https://user-images.githubusercontent.com/27722894/76100584-2c09de00-5fcd-11ea-887b-eff01586cbb9.png)

Not sure what's causing that, but the border has a different depending on what element it overlays (notice how it's different over the `UserCoverBackground` than over the background colour box) - probably a framework issue.

Reply:
This is because borders are actually drawn as part of the contained drawables. Meaning you should have a `RelativeSizeAxes = Axes.Both, Alpha = 0, AlwaysPresent = true` box at the highest level, in the masking container, to ensure a constant border (or even a border to be drawn in the first place).

Yeah this is an issue that we'll eventually need to resolve framework-side, but this is a fine solution for now.